exposed-jodatime/src/main/kotlin/org/jetbrains/exposed/sql/jodatime/DateColumnType.kt

```diff
@@ -41,11 +36,13 @@ private val MYSQL_FRACTION_DATE_TIME_AS_DEFAULT_FORMATTER by lazy {
 }
 
 private fun formatterForDateTimeString(date: String) = dateTimeWithFractionFormat(
+    date,
     date.substringAfterLast('.', "").length
 )
 
-private fun dateTimeWithFractionFormat(fraction: Int): DateTimeFormatter {
-    val baseFormat = "YYYY-MM-dd HH:mm:ss"
+private fun dateTimeWithFractionFormat(date: String, fraction: Int): DateTimeFormatter {
+    val containsDatePart = date.contains("T") || date.contains(" ")
```

Comment:
I see that it's duplicated in other datetime packages, and I believe that it works, but is it possible that `containsDatePart` is false here?

I wanted to write that if we use method inside `DateColumnType` class, and this method takes `date: String` argument, it must have date part, but now I see.

I don't expect that it's the problem of this PR, but it might be a problem in the long term. We have a `DateColumnType` class that is expected to be used with exact values from the database (actually also different values from different dialects, but it's another problem), but after that we query something like `SUBSTRING(CONVERT(NVARCHAR, TestTable."timestamptz-column", 121), 12, 15)` and expect that the column type should parse it also. And the problem is that we actually cannot say which values we expect from the database, we can rely only on what's covered by tests.

But I don't know is it easy to change with the current approach. Particularly for the time function, if the final goal is to return user the date from the whole timestamp, we can query the whole timestamp and extract time on the exposed side (instead of querying time only).

Anyway, it's just some thoughts about the potential problems with maintaining our datetime classes.